interfaces/builtin/content_test.go

```diff
@@ -354,11 +396,31 @@ slots:
 /var/snap/producer/5/export/** mrwklix,
 `
 	c.Assert(apparmorSpec.SnippetForTag("snap.consumer.app"), Equals, expected)
+
+	updateNS := apparmorSpec.UpdateNS()
+	profile0 := `  # Read-write content sharing consumer:content -> producer:content (w#0)
+  mount options=(bind, rw) /var/snap/producer/5/export/ -> /var/snap/consumer/7/import/,
+  umount /var/snap/consumer/7/import/,
+  # Writable directory /var/snap/producer/5/export
+  /var/snap/producer/5/export/ rw,
+  /var/snap/producer/5/ rw,
+  /var/snap/producer/ rw,
+  /var/snap/ rw,
+  /var/ rw,
```

Comment:
Was surprised to see that snap-update-ns needs to be able to write (ie, mkdir()) /var, /var/snap, /var/snap/producer and /var/snap/producer/export since these should already exist. Can you comment why these are needed?

Reply:
Those are automatically generated for the "make this place writable". I don't think we really need those rules but we'd have to put some extra logic and assumptions. I think that we could teach the code that certain places can be assumed to exist: e.g. `/tmp, /var/{lib,snap,cache},` etc.

Reply:
I don't think we need to go overboard here since this auto-generation of a perfect ruleset can introduce more complexity than its worth, but yes, if we can exit the loop if the path matches against a list of things like /tmp, /var/snap/SNAP_NAME/REVISION, etc (like you said), that sounds easy enough.